nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SendTrapSNMP.java

```diff
@@ -0,0 +1,269 @@
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.PORT_VALIDATOR)
+            .build();
+
+    // SNMPv1 TRAP PROPERTIES
+
+    public static final PropertyDescriptor ENTERPRISE_OID = new PropertyDescriptor.Builder()
+            .name("snmp-trap-enterprise-oid")
+            .displayName("Enterprise OID")
+            .description("Enterprise is the vendor identification (OID) for the network management sub-system that generated the trap")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .build();
+    public static final PropertyDescriptor AGENT_ADDRESS = new PropertyDescriptor.Builder()
+            .name("snmp-trap-agent-address")
+            .displayName("SNMP Trap Agent Address")
+            .description("The address where the SNMP Manager sends the trap.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .build();
+
+    public static final PropertyDescriptor GENERIC_TRAP_TYPE = new PropertyDescriptor.Builder()
+            .name("snmp-trap-generic-type")
+            .displayName("Generic Trap Type")
+            .description("Generic trap type is an integer in the range of 0 to 6. See Usage for details.")
+            .required(true)
+            .allowableValues(GENERIC_TRAP_TYPE_0, GENERIC_TRAP_TYPE_1, GENERIC_TRAP_TYPE_2, GENERIC_TRAP_TYPE_3,
+                    GENERIC_TRAP_TYPE_4, GENERIC_TRAP_TYPE_5, GENERIC_TRAP_TYPE_6)
+            .addValidator(StandardValidators.createLongValidator(0, 6, true))
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .build();
+
+    public static final PropertyDescriptor SPECIFIC_TRAP_TYPE = new PropertyDescriptor.Builder()
+            .name("snmp-trap-specific-type")
+            .displayName("Specific Trap Type")
+            .description("Specific trap type is a number that further specifies the nature of the event that generated " +
+                    "the trap in the case of traps of generic type 6 (enterpriseSpecific). The interpretation of this " +
+                    "code is vendor-specific")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .dependsOn(GENERIC_TRAP_TYPE, GENERIC_TRAP_TYPE_6)
+            .build();
+
+    public static final PropertyDescriptor TIME_STAMP = new PropertyDescriptor.Builder()
+            .name("snmp-trap-timestamp")
+            .displayName("Timestamp")
+            .description("Timestamp of the trap (unit: 0.01 second).")
+            .required(true)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .build();
+
+    // SNMPv2c/v3 TRAP PROPERTIES
```

Comment:
Is this comment necessary?

Reply:
Not necessary, but I think it's a good idea to sign the v1 and v2c/v3 properties by a comment.

Reply:
@exceptionfactory Would you please go into detail here? I'd like to grasp the underlying issue.

Reply:
@Lehel44, it is a minor concern, but since the Processor itself does not group Property Descriptors, it seems like this comment and the comment above for SNMPv1 properties do not provide much value and could easily be ignored in future changes to this processor.

Reply:
As @simonbence suggested, putting properties in intermediate collections would provide a more explicit association between properties in code without necessarily impacting the final ordering of all supported properties.

Reply:
I think organising the properties into intermediate collections adds extra complexity to the problem without solving it. If we assume that developers overlook the comments, they can overlook the collections too and new properties right to the final list. I created 4 classes to store common properties based on their functionality:

- BasicProperties
- V1TrapProperties
- V2TrapProperties
- V3SecurityProperties

In the processor classes instead of using static imports I used qualified names i.e. _V3SecurityProperties.SNMP_PRIVACY_PASSWORD_. The pattern is easy to follow and conspicuous enough not to be overlooked.